src/main/java/rx/observers/SafeCompletableSubscriber.java

```diff
@@ -52,8 +52,8 @@ public void onCompleted() {
 
     @Override
     public void onError(Throwable e) {
+        RxJavaPluginUtils.handleException(e);
```

Comment:
Done or not, the exception has to be reported to the handler. This also makes your test pass.